### ISSUE TITLE
Instructions on how to import CMasher colormaps into Tableau

### ISSUE THREAD

**Issue by rdzudzar** [1313e/CMasher#36]
Hi,
I compiled here a short "How to import CMasher colormaps into Tableau" explanation, feel free to edit it and include in your Documentation :)

Once you have your **Tableau** installed (tested with Public and Desktop version of Tableau), by default you will have **My Tableau Repository** folder inside **My Documents**.  Within this folder, you may already have or you can create Tableau _Preferences.tps_ file.

The _Preferences.tps_ file is a XML file which you can edit in text editor to input your colormaps, the empty file looks like this:

```
<?xml version='1.0'?>
<workbook>
</workbook>
```

The opening and closing preferences tags should be placed inside workbooks tags:
    
```
<?xml version='1.0'?>
<workbook>
    <preferences>
    </preferences>
</workbook>
```

Inside the preferences tags, you can place a sequence of hex numbers of your colormap, for example:

```
<?xml version='1.0'?>
<workbook>
    <preferences>    
    <color-palette name="amber" type="ordered-sequential">
        <color>#2B1619</color>
        <color>#2D171A</color>
        <color>#2E171B</color>
        <color>#30181B</color>
        <color>#31181C</color>
        ...
        <color>#EAB830</color>
        <color>#EBB931</color>
    </color-palette>
    </preferences>
</workbook>
```

For diverging colormap use `type="ordered-diverging"`, and for the cyclic colormaps use `type = "regular"`. 

Once colormaps are placed, save _Preferences.tps_ file and restart Tableau. Now you are able to use new colormaps in Tableau.    

Below I made a function that extracts all sequential, diverging and cyclic colormaps from **CMasher** and saves output as 
the _Preferences.tps_ file:

    
```
# Import packages
import cmasher as cmr
#cmr.__version__
#'1.6.0'

# In case you don't have cmasher, install it
# pip install cmasher

def create_cmasher_tableau_properties_file(cmap_range=(0.15,0.85)):
	"""
    Create Tableau parameters.tps file that contains colormaps from CMasher
    
    Parameters:
    ==========
    colormap_range: input two float numbers in range (0, 1)
    
    Returns:
    =======
    Parameters.tps -- Tableau preferences file with CMasher's colormaps
    """
    # Get colormaps
    cmr.cm.cmap_d.keys()

    # Get type of the colormaps
    cmr.cm.cmap_cd.keys()

    # We obtain names of colormaps using only those that do not end with '_r' 
    # since they are the reversed version of each colormap, and in Tableau
    # we can reverse colormaps when we use them.
    sequential_colormaps = [y for y in cmr.cm.cmap_cd['sequential'] if not y.endswith('_r')]
    diverging_colormaps = [y for y in cmr.cm.cmap_cd['diverging'] if not y.endswith('_r')]
    cyclic_colormaps = [y for y in cmr.cm.cmap_cd['diverging'] if not y.endswith('_r')]

    # Get the HEX values of each sequential colormap
    # Export them in range of: 0.15 to 0.85 -> you can export the full range if you prefer
    all_sequential_maps_hex = []
    for colormap in sequential_colormaps:
        cmp = cmr.take_cmap_colors('cmr.{}'.format(colormap), None, cmap_range=cmap_range, return_fmt='hex')
        all_sequential_maps_hex.append(cmp)

    # Get the HEX values of each diverging colormap
    all_diverging_maps_hex = []
    for colormap in diverging_colormaps:
        cmp = cmr.take_cmap_colors('cmr.{}'.format(colormap), None, cmap_range=cmap_range, return_fmt='hex')
        all_diverging_maps_hex.append(cmp)

    # Get the HEX values of each cyclic colormap
    all_cyclic_maps_hex = []
    for colormap in cyclic_colormaps:
        cmp = cmr.take_cmap_colors('cmr.{}'.format(colormap), None, cmap_range=cmap_range, return_fmt='hex')
        all_cyclic_maps_hex.append(cmp)

    # Create Tableau Preferences.tps file:
    with open("Preferences.tps", 'w') as f:
    
        print("<?xml version='1.0'?>", file=f)
        print("<workbook>", file=f)
        print("    <preferences>", file=f)
    
    	   # Get all the sequential colormaps in the format that is required in Tableau
        for i, hexes in enumerate(all_sequential_maps_hex):
            print('    <color-palette name='+'"'+'{}'.format(sequential_colormap[i])+'" ' +'type="ordered-sequential">', file=f)
            for j, col in enumerate(hexes):
                    print('        <color>'+'{}'.format(hexes[j])+'</color>', file=f)
                print('    </color-palette>', file=f)
    
    
        # Get all the cyclic colormaps in the format that is required in Tableau
        for i, hexes in enumerate(all_cyclic_maps_hex):
            print('<color-palette name='+'"'+'{}'.format(cyclic_colormap[i])+'" ' +'type="regular">', file=f)
            for j, col in enumerate(hexes):
                print('        <color>'+'{}'.format(hexes[j])+'</color>', file=f)
            print('    </color-palette>', file=f)
    
        # Get all the diverging colormaps in the format that is required in Tableau
        for i, hexes in enumerate(all_diverging_maps_hex):
            print('<color-palette name='+'"'+'{}'.format(diverging_colormap[i])+'" ' +'type="ordered-diverging">', file=f)
            for j, col in enumerate(hexes):
                print('        <color>'+'{}'.format(hexes[j])+'</color>', file=f)
            print('    </color-palette>', file=f)
    
        print("    </preferences>", file=f)
        print("</workbook>", file=f)

if __name__ == "__main__":
    #Change cmap_range as required from 0 to 1.
    cmap_range = (0.15, 0.85)
    create_cmasher_tableau_properties_file(cmap_range)
```

An example that shows imported **CMasher** colormaps in Tableau:

![Tableau_cmaps](https://user-images.githubusercontent.com/33647339/116078658-3cea4700-a6da-11eb-98d7-f9217812f167.png)
:

Cheers

**Comment by 1313e** [comment]
Thank you very much, @rdzudzar.
Always great to see people using CMasher in different ways than just Python.
I will have a look at this today.

**Comment by rdzudzar** [comment]
Tested your code for producing the `Preferences.tps` file, all works well. Here is the updated image:
![Tableau_cmaps](https://user-images.githubusercontent.com/33647339/116365563-b86a0680-a848-11eb-9442-c009490abac5.png)

**Comment by 1313e** [comment]
This was added to CMasher in ebba3c9 and will be available in v1.6.1.
